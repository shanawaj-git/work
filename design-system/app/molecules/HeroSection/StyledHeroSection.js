import styled from 'styled-components';

import { colorSelector } from 'themes';

export const HeroSectionWrapper = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
  .title-img {
    width: 100%;
    height: auto;
  }

  .title-container {
    margin-top: 20px;
    width: 100%;
    width: 100%;
  }

  .title-logo {
    margin-top: 5px;
  }

  .headline {
    color: ${colorSelector('font-tertiary')};
    margin-top: 16px;
    margin-left: 20px;
    margin-right: 20px;
  }
`;
