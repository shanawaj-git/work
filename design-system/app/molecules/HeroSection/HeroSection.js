import React from 'react';
import PropTypes from 'prop-types';

import Typography, { TypographyType } from 'atoms/Typography';
import { HeroSectionWrapper } from './StyledHeroSection';

const HeroSection = props => {
  const { className, image, title, headLine, titleLogo } = props;

  return (
    <div className={className}>
      <HeroSectionWrapper>
        <img src={image} className="title-img" alt={title} />
        <div className="title-container">
          <Typography typographyType={TypographyType.LARGE_TITLE}>
            {title}
          </Typography>
          {titleLogo && (
            <img alt="temp" className="title-logo" src={titleLogo} />
          )}
        </div>
        <Typography
          className="headline"
          typographyType={TypographyType.SUB_HEAD}
        >
          {headLine}
        </Typography>
      </HeroSectionWrapper>
    </div>
  );
};

export default HeroSection;

HeroSection.propTypes = {
  className: PropTypes.string,
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  headLine: PropTypes.string.isRequired,
  titleLogo: PropTypes.string,
};
