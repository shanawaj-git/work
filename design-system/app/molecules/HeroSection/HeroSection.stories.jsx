import React from 'react';

import StoryWrapper from 'molecules/utility/StoryWrapper';
import HeroSection from './HeroSection';

export default {
  title: 'Components/Containers/Cards/HeroSection',
  component: HeroSection,
  argTypes: {
    themeType: {
      control: { type: 'select', options: ['DEFAULT', 'MPC'] },
    },
    className: { table: { disable: true } },
  },
};

const Template = args => (
  <StoryWrapper themeVariant={args.themeType}>
    <HeroSection {...args} />
  </StoryWrapper>
);

export const HeroSectionComponent = Template.bind({});

const icon = require('images/header-image.png');
const titleLogo = require('images/next-health-logo-primary.svg');

HeroSectionComponent.args = {
  image: icon,
  title: 'Welcome to',
  titleLogo,
  headLine:
    'Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
};
