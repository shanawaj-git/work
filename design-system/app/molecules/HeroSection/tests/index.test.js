import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';

import HeroSection from '../index';

const shallowRenderer = new ShallowRenderer();

const headerImage = require('images/header-image.png');
const logoImage = require('images/next-health-logo-primary.svg');

const defaultProps = {
  image: headerImage,
  title: 'Order placed!',
  subTitle: 'Your order is placed and will be dispatched to you shortly.',
};

describe('HeroSection.js', () => {
  describe('index', () => {
    it('Should import succesfully', () => {
      const tree = renderer.create(<HeroSection {...defaultProps} />);
      expect(tree).toBeTruthy();
    });

    it('Should display header image on top of main title and subtitle ', () => {
      shallowRenderer.render(<HeroSection {...defaultProps} />);
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });

    it('Should display the titleLogo when passed', () => {
      shallowRenderer.render(
        <HeroSection {...defaultProps} titleLogo={logoImage} />,
      );
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });

    it('Should not display the titleLogo when not passed', () => {
      shallowRenderer.render(
        <HeroSection {...defaultProps} titleLogo={null} />,
      );
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });
  });
});
