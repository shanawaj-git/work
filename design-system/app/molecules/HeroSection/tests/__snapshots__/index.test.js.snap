// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection.js index Should display header image on top of main title and subtitle  1`] = `
<div>
  <ForwardRef(StyledHeroSection__HeroSectionWrapper)>
    <img
      alt="Order placed!"
      className="title-img"
      src="IMAGE_MOCK"
    />
    <div
      className="title-container"
    >
      <Typography
        typographyType="largeTitle"
      >
        Order placed!
      </Typography>
    </div>
    <Typography
      className="headline"
      typographyType="subHead"
    />
  </ForwardRef(StyledHeroSection__HeroSectionWrapper)>
</div>
`;

exports[`HeroSection.js index Should display the titleLogo when passed 1`] = `
<div>
  <ForwardRef(StyledHeroSection__HeroSectionWrapper)>
    <img
      alt="Order placed!"
      className="title-img"
      src="IMAGE_MOCK"
    />
    <div
      className="title-container"
    >
      <Typography
        typographyType="largeTitle"
      >
        Order placed!
      </Typography>
      <img
        alt="temp"
        className="title-logo"
        src="IMAGE_MOCK"
      />
    </div>
    <Typography
      className="headline"
      typographyType="subHead"
    />
  </ForwardRef(StyledHeroSection__HeroSectionWrapper)>
</div>
`;

exports[`HeroSection.js index Should not display the titleLogo when not passed 1`] = `
<div>
  <ForwardRef(StyledHeroSection__HeroSectionWrapper)>
    <img
      alt="Order placed!"
      className="title-img"
      src="IMAGE_MOCK"
    />
    <div
      className="title-container"
    >
      <Typography
        typographyType="largeTitle"
      >
        Order placed!
      </Typography>
    </div>
    <Typography
      className="headline"
      typographyType="subHead"
    />
  </ForwardRef(StyledHeroSection__HeroSectionWrapper)>
</div>
`;
